python/lsst/ip/diffim/subtractImages.py

```diff
@@ -772,26 +785,20 @@ def _sourceSelector(self, sources, mask):
             If there are too few sources to compute the PSF matching kernel
             remaining after source selection.
         """
-        flags = np.ones(len(sources), dtype=bool)
-        for flag in self.config.badSourceFlags:
-            try:
-                flags *= ~sources[flag]
-            except Exception as e:
-                self.log.warning("Could not apply source flag: %s", e)
-        signalToNoise = sources.getPsfInstFlux()/sources.getPsfInstFluxErr()
-        sToNFlag = signalToNoise > self.config.detectionThreshold
-        flags *= sToNFlag
-        sToNFlagMax = signalToNoise < self.config.detectionThresholdMax
-        flags *= sToNFlagMax
-        flags *= self._checkMask(mask, sources, self.config.excludeMaskPlanes)
-        selectSources = sources[flags].copy(deep=True)
+
+        selected = self.sourceSelector.selectSources(sources).selected
+        nInitialSelected = np.count_nonzero(selected)
+        selected *= self._checkMask(mask, sources, self.config.excludeMaskPlanes)
+        nSelected = np.count_nonzero(selected)
+        self.log.info("%i candidate sources rejected from template mask", nInitialSelected - nSelected)
```

Comment:
Instead of "rejected from template mask", perhaps "rejected due to regions in template with one or more excludeMaskPlanes" ?